src/wavm/WAVMWasmModule.cpp

```diff
@@ -969,21 +989,23 @@ int32_t WAVMWasmModule::executeAsPthread(uint32_t stackTop,
     int argsPtr = std::stoi(msg->inputdata());
     std::vector<IR::UntaggedValue> invokeArgs = { argsPtr };
 
-    WasmThreadSpec spec = {
-        .contextRuntimeData = getContextRuntimeData(executionContext),
-        .func = funcInstance,
-        .funcArgs = invokeArgs.data(),
-        .stackTop = stackTop,
-    };
+    Runtime::ContextRuntimeData* contextRuntimeData =
+      getContextRuntimeData(executionContext);
+
+    // Set up the context
+    Runtime::Context* threadContext =
+      createThreadContext(stackTop, contextRuntimeData);
 
     // Record the return value
-    I64 returnValue = executeThreadLocally(spec);
```

Comment:
The `executeThreadLocally` function was no longer useful, hence the slight refactor of both the OpenMP and pthread bits.